cpp/src/arrow/gpu/cuda_context.h

```diff
@@ -76,6 +96,13 @@ class ARROW_EXPORT CudaContext : public std::enable_shared_from_this<CudaContext
   /// \return Status
   Status Allocate(int64_t nbytes, std::shared_ptr<CudaBuffer>* out);
 
+  /// \brief Create a view of CUDA memory on GPU device of this context
```

Comment:
Maybe worth commenting here that the created CudaBuffer cannot retain ownership of the memory as in the Cython code comment

Reply:
Sure. Done.